### DISCUSSION TITLE
Some questions refering to Custom Collections

### DISCUSSION THREAD

**Discussion by EsQueEl-Fella** [trimble-oss/dba-dash#776] [Q&A] [answered]
Hi David,

I have been waiting for this feature for some times and now that the feature is there I am very excited to set it up. I have been checking your Video this regard many times and followed you instructions step by step but unfortunately I still am puzzling around and it is unclear to me how things are working or maybe I am totally missing the point. Anyways I decided to open up this thread to get some better understanding.

Sorry if the questions below might be a bit off or trivial but I really I am having difficulty to understand the process. I am not quite sure but I also think I have found a small (but important) bug in the Application this regard. If this is true or not, can be verified after I get a better understanding

**Question 1**: 
**In which database does DBADash Agent look for my Stored Procedures?**
- DBADash Repository Database? (Then referring to this case in your HowTo/Sample Video is missing.)
- A specific User Database which should exist on every Instance? Like an “Admin” or “Tools” Database? (Then referring to this case in your HowTo/Sample Video is missing.)
- The Default Database for DBADash Agent? (... !!!??? ...)
- Any Database on any Instance? (Somehow I can't think of a case if this option would be working if we wanted to setup Custom Collections for ALL instances!) 

**Question 2**:
**Where is the call to my Stored Procedure?**
I wasn’t able to clearly identify how my custom SP is going to be called. In your (sp_Blitz) example I see a TYPE, A TABLE and a Wrapper PRODECURE but I simply can’t find an EXECUTE to (sp_Blitz or) My Stored Procedure ! Am I missing something here? Please be a bit Detailed

**A possible Bug in the application (which I mentioned above):**
If I try to setup a Custom Collection for ALL Instances, first I need to connect to an Instance and here of course I can (optionally) provide a database on that instance.
-	Why Optionally?
If I try to setup a Custom Collection ONLY for one Instance, I am not able to specify any database because DBADash connects to the Default database, “master” ! (My DBADash Agent is “sysadmin” for simplicity)
-	Why? (I believe this is a bug.)

I certainly don’t save my custom SPs in the “master” database which in return means I am also not able to choose any one of them and this makes setting up the Custom Collection Feature for individual instances impossible.

I hope now one can follow why I am asking the above questions.

**Comment by DavidWiseman** [comment, marked as answer]
Question 1

The custom collection stored procedure is created in the database that DBA Dash connects to.  This refers to procedure that is going to collect the data from your monitored instances. e.g. sp_Blitz or whatever custom script you want to run.
Typically the agent will connect to the master database so you can either create the procedure there or alter the connection string for the source connection to specify a different default database.

Question 2

The script generated when you click the "Get Script" button is to be run in the repository database only.  There isn't anything here to run sp_Blitz as that will be run by the agent.  The purpose of the script is to provide a place in the repository DB to store the data and a mechanism to handle the data import.  

If you setup a collection for all instances you are prompted for a connection.  The only purpose of this is to populate the stored procedure drop down list.  It's also used when you click the Get Script button.  It doesn't influence where the agent will collect the data from in any way.  

If you don't want to create your custom collection SPs in master, you can change the source connections to have a different default database.

Hope this helps

**Reply by EsQueEl-Fella** [reply]
Hi David, 

the answers to my questions were the missing pieces of the puzzle. Thanks you very much. I also do appreciate your fast reply. Probably (or even definitely) if I had created my connection strings with a default database originally, I would never even came across these questions.

**Comment by DavidWiseman** [comment]
PS.  The documentation for DBA Dash is also open source if you are interested in improving the documentation.  e.g. 
https://github.com/trimble-oss/dba-dash-docs/blob/main/content/en/docs/help/custom-collections/index.md

**Reply by EsQueEl-Fella** [reply]
I do like to be of help and provide some technical contribution but I am afraid my current knowledge of application and how things are working are not at the proper level. I hope one day I am confident enough to do that but until then I surly try to provid information, user experience and testings and let you decide and evaluate the reported infromation. 

Here one example you could review:
Some of the links in the doc are pointing to nowhere :D. There is alos one more I could which is not in the image attached
![grafik](https://github.com/trimble-oss/dba-dash/assets/40255900/a075bb87-125b-4d1c-911b-0e46c7eb6221)

**Comment by DavidWiseman** [comment]
The links don't work on GitHub because they get redirected.  Once the website is built the links work:
https://dbadash.com/docs/help/custom-collections/

Thanks for reporting issues - that's also useful.  😊
